### PR TITLE
Bump mockito from 5.14.1 to 5.14.2, objenesis from 3.2 to 3.3 and bytebuddy from 1.15.4 to 1.15.10

### PR DESCRIPTION
**Pull request by cwperks**
### Description

Bumps the version of mockito from 5.14.1 to 5.14.2, objenesis from 3.2 to 3.3 and bytebuddy from 1.15.4 to 1.15.10

### Check List
- [ ] Functionality includes testing.
- [ ] API changes companion pull request [created](https://github.com/opensearch-project/opensearch-api-specification/blob/main/DEVELOPER_GUIDE.md), if applicable.
- [ ] Public documentation issue/PR [created](https://github.com/opensearch-project/documentation-website/issues/new/choose), if applicable.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: gradle/libs.versions.toml [hunk 1]
```diff
@@ -62,8 +62,8 @@ bouncycastle="1.78"
 randomizedrunner  = "2.7.1"
 junit             = "4.13.2"
 hamcrest          = "2.1"
-mockito           = "5.14.1"
-objenesis         = "3.2"
+mockito           = "5.14.2"
+objenesis         = "3.3"
 bytebuddy         = "1.15.4"
```
[the remark this answers is not in the document]
Comment (cwperks) [reply gradle/libs.versions.toml]: Updated to 1.15.10

[the remark this answers is not in the document]
Comment (cwperks) [reply gradle/libs.versions.toml]: @reta mockito is showing bytebuddy 1.15.4 for v5.14.2 of mockito: https://github.com/mockito/mockito/blob/78348597236d54a84843f454936b8440b006f773/gradle/libs.versions.toml#L2

[the remark this answers is not in the document]
Comment (reta) [reply gradle/libs.versions.toml]: Should be non-issue I believe, it is good to have the latest one, thanks @cwperks !

**Review by reta — APPROVED** [review]

**Comment by cwperks** [comment]
> java.security.AccessControlException: access denied ("java.lang.RuntimePermission" "createClassLoader")

**Comment by reta** [comment]
> > java.security.AccessControlException: access denied ("java.lang.RuntimePermission" "createClassLoader")

We may need to update `https://github.com/opensearch-project/OpenSearch/blob/main/libs/common/src/test/resources/org/opensearch/bootstrap/test.policy` as well (new permission)

**Comment by cwperks** [comment]
Added the missing perms

**Review by reta — APPROVED** [review]

**Pull request merged**